Mage.Sets/src/mage/cards/i/IntrudeOnTheMind.java

```diff
@@ -0,0 +1,159 @@
+
+        TargetCard target = new TargetCard(0, cards.size(), Zone.LIBRARY, filter);
+        List<Card> pile1 = new ArrayList<>();
+        controller.choose(Outcome.Neutral, cards, target, source, game);
+        target.getTargets()
+                .stream()
+                .map(game::getCard)
+                .filter(Objects::nonNull)
+                .forEach(pile1::add);
+        cards.removeIf(target.getTargets()::contains);
+        List<Card> pile2 = new ArrayList<>();
+        pile2.addAll(cards.getCards(game));
+
+        Player opponent = getOpponent(controller, game, source);
+        boolean choice = opponent.choosePile(outcome, "Choose a pile to put into hand.", pile1, pile2, game);
+
+        Zone pile1Zone = choice ? Zone.HAND : Zone.GRAVEYARD;
+        Zone pile2Zone = choice ? Zone.GRAVEYARD : Zone.HAND;
+
+        game.informPlayers("Pile 1, going to " + pile1Zone + ": " + (pile1.isEmpty() ? " (none)" :
+                pile1.stream().map(MageObject::getName).collect(Collectors.joining(", "))));
+        cards.clear();
+        cards.addAllCards(pile1);
+        controller.moveCards(cards, pile1Zone, source, game);
+
+        game.informPlayers("Pile 2, going to " + pile2Zone + ": " + (pile2.isEmpty() ? " (none)" :
+                pile2.stream().map(MageObject::getName).collect(Collectors.joining(", "))));
+        cards.clear();
+        cards.addAllCards(pile2);
+        controller.moveCards(cards, pile2Zone, source, game);
+
+        Token token = new ThopterColorlessToken2();
+        token.putOntoBattlefield(1, game, source);
+
+        allCards.retainZone(Zone.GRAVEYARD, game);
+        int count = allCards.size();
+        if (count <= 0) {
+            return true;
+        }
+        for (UUID tokenId : token.getLastAddedTokenIds()) {
+            Permanent permanent = game.getPermanent(tokenId);
+            if (permanent == null) {
+                continue;
+            }
+            permanent.addCounters(CounterType.P1P1.createInstance(count), source.getControllerId(), source, game);
+        }
+        return true;
+    }
+
+    private static Player getOpponent(Player controller, Game game, Ability source) {
+        Player opponent;
+        Set<UUID> opponents = game.getOpponents(source.getControllerId());
+        if (opponents.isEmpty()) {
+            return null;
+        }
+        if (opponents.size() == 1) {
+            opponent = game.getPlayer(opponents.iterator().next());
+        } else {
+            Target targetOpponent = new TargetOpponent(true);
+            controller.chooseTarget(Outcome.Neutral, targetOpponent, source, game);
+            opponent = game.getPlayer(targetOpponent.getFirstTarget());
+            if (opponent != null) {
+                game.informPlayers(controller.getLogName() + " chose " + opponent.getLogName() + " to choose the piles");
+            } else {
+                game.informPlayers(controller.getLogName() + " chose nothing" + " to choose the piles");
```

Comment:
(idk why bother with info message on failed choice, shouldn't happen in normal usage)

Reply:
ah you still need to null check it though